### PR TITLE
[HotReload] Skips Browser refresh middleware when running against an unsupported target framework version

### PR DESCRIPTION
**Pull request by javiercn**
* Checks the target framework of the project and emits a warning message if the version is not supported.

### CODE REVIEW

File: src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs [hunk 1]
```diff
@@ -38,6 +40,13 @@ public async ValueTask ProcessAsync(DotNetWatchContext context, CancellationToke
                     _reporter.Verbose("Unable to determine if this project is a webapp.");
                     return;
                 }
+                else if (!IsSupportedVersion(context.ProjectGraph))
+                {
+                    _reporter.Warning(
+                        "Skipping configuring browser-refresh middleware since the target framework version is not supported." +
+                        " For more information see 'https://aka.ms/dotnet/watch/unsupported-version'.");
```
Comment (javiercn) [inline src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs:47]: @mkArtakMSFT does this URL work for you (we'll have to create it and point it to the docs).

Comment (mkArtakMSFT) [reply src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs:47]: Thanks @javiercn . Can you please change the link to this one instead:
https://aka.ms/dotnet/watch/unsupported-tfm

**Comment by mkArtakMSFT** [comment]
@marcpopMSFT, @mmitche which branch should we target this fix to make sure it rolls out with the upcoming 8.0.1 release?

**Comment by baronfel** [comment]
@mkArtakMSFT release/8.0.1xx is the servicing branch for 8.x currently

**Comment by mkArtakMSFT** [comment]
Thank you, @baronfel.
@javiercn please create the backport PR when this is ready, and I'll send that to Tactics on Monday.

**Comment by javiercn** [comment]
First it should go in main, then I'll backport it

File: src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs (hunk 1 above)
Comment (javiercn) [reply src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs:47]:
```suggestion
                        " For more information see 'https://aka.ms/dotnet/watch/unsupported-tfm'.");
```

**Review by MackinnonBuck — APPROVED** [review]
Looks great, just a small question about the warning message.

File: src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs [hunk 2]
```diff
@@ -10,6 +10,8 @@ namespace Microsoft.DotNet.Watcher.Tools
 {
     internal sealed class BrowserRefreshFilter : IWatchFilter, IAsyncDisposable
     {
+        // This needs to be in sync with the version BrowserRefreshMiddleware is compiled against.
```
Comment (halter73) [inline src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs:13]: Should we add a comment about this here?

https://github.com/dotnet/sdk/blob/45e21d9c56cc7d3c8b6cb2db53d0c63dfe47e1bb/src/BuiltInTools/BrowserRefresh/Microsoft.AspNetCore.Watch.BrowserRefresh.csproj#L3-L4

I figure that's where people changing the BrowserRefresh TFM would likely see the comment.

File: src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs [hunk 3]
```diff
@@ -66,6 +75,24 @@ public async ValueTask ProcessAsync(DotNetWatchContext context, CancellationToke
             }
         }
 
+        private bool IsSupportedVersion(ProjectGraph context)
+        {
+            if (context.GraphRoots.FirstOrDefault() is not { } projectNode)
+            {
+                return false;
+            }
+            if (projectNode.ProjectInstance.GetPropertyValue("_TargetFrameworkVersionWithoutV") is not string targetFrameworkVersion)
```
Comment (tmat) [inline src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshFilter.cs:84]: Can we avoid using private properties (prefixed with `_`)?

**Pull request merged**